massa-final-state/src/final_state.rs

```diff
@@ -235,12 +189,96 @@ impl FinalState {
             .expect("Cycle history should not be empty in snapshot!")
             .clone();
 
-        // Reset the cycle_history and populate it with new cycle
-        self.pos_state.cycle_history = Vec::new().into();
-        self.pos_state
-            .create_new_cycle_for_snapshot(&latest_consistent_cycle_info, end_slot);
+        if current_slot_cycle == end_slot_cycle {
```

Comment:
move the two bodies of this if-statement into two helper functions, the names of which help self-document


(e.g. `interpolate_single_cycle` and `interpolate_multiple_cycles`, or something along those lines)

Reply:
Done in refactor 1405887a97bd4fbf5444e4b09f5cf2da48919a27